File: lab-2/src/moves/RockSlide.java
package moves;
import ru.ifmo.se.pokemon.*;

public class RockSlide extends PhysicalMove{
    public RockSlide(){
        super(Type.ROCK, 75, 90);
    }

    protected void applyOppEffects(Pokemon p){
        if(Math.random()<0.3){
            Effect.flinch(p);
        }
    }

    protected String describe(){
        return "is using Rock Slide";
    }
}

File: lab-2-kotlin/src/moves/RockSlide.kt
package moves

import ru.ifmo.se.pokemon.*

class RockSlide : PhysicalMove(Type.ROCK, 75.0, 90.0) {
    override fun applyOppEffects(p: Pokemon) {
        if (Math.random() < 0.3) {
            Effect.flinch(p)
        }
    }

    override fun describe(): String {
        return "is using Rock Slide"
    }
}